core/internal/cltest/mocks.go

```diff
@@ -213,16 +213,19 @@ func (mock *EthMock) Call(result interface{}, method string, args ...interface{}
 // Call, falling back to reflection if the values dont support the required
 // encoding interfaces
 func assignResult(result, response interface{}) error {
-	unmarshaler, uok := result.(encoding.TextUnmarshaler)
-	marshaler, mok := response.(encoding.TextMarshaler)
-
-	if uok && mok {
-		bytes, err := marshaler.MarshalText()
-		if err != nil {
-			return err
+	if unmarshaler, ok := result.(encoding.TextUnmarshaler); ok {
```

Comment:
**Review note:** this change makes `assignResult` capable of dealing with a `string` or `[]byte` value for `response`.  Neither of those conform to `encoding.TextUnmarshaler`, but this is trivial to support.